app/walletButton.tsx

```diff
@@ -9,128 +9,177 @@ import {
-          style={{ maxWidth: 360 }}
-        >
-          <Button
-            className="w-full flex items-center text-left mt-2"
-            onClick={onConnectBitcoin}
-          >
-            <Image
-              src="/logo-btc.svg"
-              className="icon mr-2"
-              alt="Bitcoin Logo"
-              height={24}
-              width={24}
-            />{" "}
-            Bitcoin
-          </Button>
-          <Button
-            className="w-full flex items-center text-left mt-2"
-            onClick={onConnectEthereum}
-          >
-            <Image
-              src="/logo-eth.svg"
-              className="icon mr-2"
-              alt="Bitcoin Logo"
-              height={24}
-              width={24}
-            />{" "}
-            Ethereum
-          </Button>
-        </Modal>
-      ) : null}
       {modal && modal.type == "wallet" ? (
         <ModalWallet
-          wallet={wallet!}
           onClose={() => setModal(undefined)}
-          setWallet={setWallet}
         />
       ) : null}
     </>
   );
 }
 
 function ModalWallet({
-  wallet,
   onClose,
-  setWallet,
 }: {
-  wallet: Wallet;
   onClose: () => void;
-  setWallet: (_: undefined | object) => void;
 }) {
-  const [balance, setBalance] = useState(0);
+  const [wallets, setWallets] = useAtom(atomWallet);
+  const { address: ethAddress, isConnected: isEthConnected } = useAccount();
+  const { disconnect } = useDisconnect();
+  const { data: ethBalance } = useBalance({ address: ethAddress });
+  const { openConnectModal } = useConnectModal();
+
+  const [btcBalance, setBtcBalance] = useState(0);
 
   useEffect(() => {
-    (async () => {
-      setBalance(await wallet.getBalance());
-    })();
-  }, [wallet]);
+    async function fetchBitcoinBalance() {
+      if (wallets.bitcoin) {
+        const balance = await wallets.bitcoin.getBalance();
+        setBtcBalance(balance);
+      }
+    }
+    fetchBitcoinBalance();
+  }, [wallets.bitcoin]);
 
-  function onDisconnect() {
-    onClose();
-    setWallet(undefined);
+  const updateEthereumWallet = useCallback(() => {
+    if (isEthConnected && ethAddress) {
+      setWallets(prev => ({
+        ...prev,
+        ethereum: {
+          chain: "ethereum",
+          symbol: "ETH",
+          address: ethAddress,
+          getBalance: async () => Number(ethBalance?.formatted || 0),
+        },
+      }));
+    } else {
+      setWallets(prev => {
+        const newWallets = { ...prev };
+        delete newWallets.ethereum;
+        return newWallets;
+      });
+    }
+  }, [isEthConnected, ethAddress, ethBalance, setWallets]);
+
+  useEffect(() => {
+    updateEthereumWallet();
+  }, [updateEthereumWallet]);
+
+  async function onConnectBitcoin() {
+    try {
+      const bitcoinWallet = await bitcoinConnectInjected();
+      setWallets(prev => ({ ...prev, bitcoin: bitcoinWallet }));
+    } catch (error) {
+      console.error("Failed to connect Bitcoin wallet:", error);
```

Comment:
we should display something to the user

Reply:
I've added a task on linear for error handling in general.

Reply:
Can you link to ticket?

Reply:
[YLD-66: App wide error handler with error provider + errormodal](https://linear.app/project-chaos/issue/YLD-66/app-wide-error-handler-with-error-provider-errormodal)